src/main/java/com/internship/backend/service/UserService.java

```diff
@@ -4,119 +4,123 @@
 import com.internship.backend.exceptions.UserAlreadyExistsException;
 import com.internship.backend.exceptions.UserDoesNotExistException;
 import com.internship.backend.model.Authority;
+import com.internship.backend.model.Reservation;
 import com.internship.backend.model.Users;
 import com.internship.backend.repository.AuthorityRepository;
+import com.internship.backend.repository.ReservationRepository;
 import com.internship.backend.repository.UserRepository;
-import jakarta.transaction.Transactional;
-import lombok.RequiredArgsConstructor;
-import org.apache.commons.lang3.ObjectUtils;
+
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.security.core.userdetails.MapReactiveUserDetailsService;
 import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.IdGenerator;
 
-import java.util.HashSet;
-import java.util.Iterator;
-import java.util.List;
-import java.util.Set;
-
-
+import java.util.*;
 @Service
-@RequiredArgsConstructor
 public class UserService {
 
     @Autowired
     private UserRepository userRepository;
 
     @Autowired
-    private PasswordEncoder passwordEncoder;
-
+    private IdGeneratorService idGeneratorService;
 
     @Autowired
     private AuthorityRepository authorityRepository;
-    private MapReactiveUserDetailsService reactiveUserDetailsService;
 
-    @Transactional
-    public List<Users> getAllUsersProcedure(){
-        return userRepository.getAllUsersProcedure();
-    }
+    @Autowired
+    private PasswordEncoder passwordEncoder;
 
-    @Transactional
-    public void deleteUserByIdProcedure(Integer userId){
-        authorityRepository.deleteByUserId(userId);
-        userRepository.deleteUserByIdProcedure(userId);
-    }
+    Logger logger = LoggerFactory.getLogger(UserService.class);
+    @Autowired
+    private ReservationRepository reservationRepository;
 
+    @Transactional
     public Users register(Users user) throws UserAlreadyExistsException {
-        if (userRepository.findByUsername(user.getUsername()) != null) {
+        if (userRepository.findByUsername(user.getUsername()).isPresent()) {
             throw new UserAlreadyExistsException("Username already exists");
         }
-        if (userRepository.findByEmail(user.getEmail()) != null) {
+        if (userRepository.findByEmail(user.getEmail()).isPresent()) {
             throw new UserAlreadyExistsException("Email already exists");
         }
+
+        user.setId(idGeneratorService.getCurrentId());
         String hashPassword = passwordEncoder.encode(user.getPassword());
         user.setPassword(hashPassword);
 
         Users savedUser = userRepository.save(user);
```

Comment:
can use var instead of Users ( var savedUser = userRepository.save(user)). Applicable in other places as well